am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -35,6 +44,83 @@ void whenRevokingResourceAccessResourceAccessRemovedFromDatabase() {
         assertThat(countResourcesById(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceAccessOnRootResourceAccessRemovedFromDatabase() {
+        ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
+        ams.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(ACCESSOR_ID)
+            .accessType(ACCESS_TYPE)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute("/")
+            .securityClassification(SECURITY_CLASSIFICATION)
+            .build());
+
+        assertThat(countResourcesById(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenRevokingResourceAccessOnSingleNestedAttributeResourceAccessRemovedFromDatabase() {
+        ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
+        ams.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(ACCESSOR_ID)
+            .accessType(ACCESS_TYPE)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute("/test/childTest")
+            .securityClassification(SECURITY_CLASSIFICATION)
+            .build());
+
+        assertThat(countResourcesById(resourceId)).isEqualTo(0);
```

Comment:
That test shows it does not really work. When access is granted to the whole document and only certain attribute access is revoked it an access to the whole doc should remain.

Your logic simply removes everything from DB at any time.

Reply:
👍 tested and works for multiple scenarios.